src/audio/component.c

```diff
@@ -506,6 +516,46 @@ int comp_copy(struct comp_dev *dev)
 	return ret;
 }
 
+#ifdef CONFIG_SOF_TELEMETRY_PERFORMANCE_MEASUREMENTS
+void comp_init_performance_data(struct comp_dev *dev)
+{
+	struct perf_data_item_comp *item = dev->perf_data.perf_data_item;
+
+	if (item)
+		perf_data_item_comp_init(item, dev->ipc_config.id, 0);
+}
+
+/* returns true if budget violation occurred */
+static bool update_peak_of_measured_cpc(struct comp_dev *dev, size_t measured_cpc)
+{
+	if (measured_cpc <= dev->perf_data.peak_of_measured_cpc)
+		return false;
+	dev->perf_data.peak_of_measured_cpc = measured_cpc;
+	if (measured_cpc <= dev->cpc)
+		return false;
+	return true;
+}
+
+bool comp_update_performance_data(struct comp_dev *dev, uint32_t cycles_used)
+{
+	struct perf_data_item_comp *item = dev->perf_data.perf_data_item;
+
+	if (perf_meas_get_state() == IPC4_PERF_MEASUREMENTS_STARTED) {
+		/* we divide by ibs so we need to check if its set */
+		if (item && dev->ibs != 0) {
+			item->total_iteration_count++;
```

Comment:
how often does this run? If this runs every millisecond, then in a couple of days this will overflow and the division below will divide by zero

Reply:
Yeah it may as well be every ms. Doubt it will be used for a couple of days , but still I should add some check here, true.